cmd/lotus-storage-miner/sectors.go

```diff
@@ -524,7 +527,8 @@ var sectorsExtendCmd = &cli.Command{
 				} else {
 					added := false
 					for exp := range es {
-						if withinTolerance(exp, newExp) {
+						// newExp < exp, can't extend
+						if withinTolerance(exp, newExp) && newExp >= exp {
```

Comment:
Nice catch, thank you! The optimal thing to do would be to repack the extensions, replacing `exp` with the smaller `newExp` in this case, but I don't think it's too important.